src/helper.py

```diff
@@ -88,3 +97,99 @@ def generate_source_code(
         highlights=highlights,
         transformation_display_name=transformation_display_name,
     )
+
+
+def collect_assets_fullname(
+    collibra_instance: str,
+    username: str,
+    password: str,
+    type_id: Optional[str] = None,
+    domain_id: Optional[str] = None,
+    name: Optional[str] = None,
+) -> List[AssetFullnameDomain]:
+    """
+    Helper function that collect assets fullname from Collibra
+
+    :param collibra_instance: Collibra instance name
+    :type collibra_instance: str
+    :param username: Collibra username
+    :type username: str
+    :param password: Collibra user's password
+    :type password: str
+    :param type_id: Optional parameter - Asset type ID
+    :type type_id: str
+    :param domain_id: Optional parameter - Collibra domain ID
+    :type domain_id: str
+    :param name: Optional parameter - Assets name
+    :type name: str
+    :returns: list of dictionary
+    :rtype: list
+    """
+
+    def get_assets_fullname_from_collibra() -> List[AssetFullnameDomain]:
+        fullnames = []
+        auth = HTTPBasicAuth(username=username, password=password)
+        cursor = urllib.parse.quote("")
+        limit = 1000
+        attempt = 1
+        base_path = f"https://{collibra_instance}.collibra.com/rest/2.0/assets?limit={limit}"
+
+        if domain_id:
+            base_path = base_path + f"&domainId={domain_id}"
+        if type_id:
+            base_path = base_path + f"&typeIds={type_id}"
+        if name:
+            base_path = base_path + f"&name={urllib.parse.quote(name)}"
+
+        while cursor is not None:
+            query_path = base_path + f"&cursor={cursor}"
+            try:
+                logging.debug(f"Sending GET {query_path}")
+                ret = requests.get(query_path, auth=auth)
+            except NameResolutionError as e:
+                raise e
+            except Exception as e:
+                logging.warning(f"GET {query_path} failed with\n{e}")
+                attempt += 1
+            else:
+                if ret.status_code == 200:
+                    logging.debug(f"Response received for GET {query_path}: {ret.status_code}")
+                    attempt = 1
+                    result = json.loads(ret.text)
+                    cursor = result.get("nextCursor")
+                    for entry in result.get("results", []):
+                        fullnames.append(
+                            AssetFullnameDomain(
```

Comment:
just a question here, what is the point of having default empty string as that does not help?